docs/quickstarts/javascript.mdx

```diff
@@ -1,183 +1,298 @@
+      - [`<SignIn />`](/docs/references/javascript/clerk/sign-in): renders a user interface for signing in.
+      - [`<UserButton />`](/docs/references/javascript/clerk/user-button): shows the avatar from the account the user is signed in with. When clicked, it opens a dropdown menu with options to sign out.
 
-<Images width={496} height={564} src="/docs/images/ui-components/component-user_button.svg" alt="User Button Example" />
+      <Tabs items={["index.js", "index.html"]}>
+        <Tab>
+          ```js filename="main.js"
+          import Clerk from "@clerk/clerk-js";
 
-<InjectKeys>
+          const clerkPubKey = import.meta.env.VITE_CLERK_PUBLISHABLE_KEY;
 
-<CodeBlockTabs type="npm-script" options={['NPM Module', 'window.Clerk']}>
-  ```typescript {14}
-  import Clerk from '@clerk/clerk-js';
+          if (!clerkPubKey) {
+            throw new Error("Add your VITE_CLERK_PUBLISHABLE_KEY to .env file");
+          }
 
-  document.querySelector<HTMLDivElement>('#app')!.innerHTML = `
-    <div
-      id="user-button"
-    ></div>
-  `;
+          const clerk = new Clerk(clerkPubKey);
+          await clerk.load();
 
-  const userButtonComponent = document.querySelector<HTMLDivElement>('#user-button')!;
+          if (clerk.user) {
+            document.getElementById("app").innerHTML = `
+              <div id="user-button"></div>
+            `;
 
-  const clerk = new Clerk(`{{pub_key}}`);
-  await clerk.load();
+            const userButtonDiv =
+              document.getElementById("user-button");
 
-  clerk.mountUserButton(userButtonComponent);
-  ```
+            clerk.mountUserButton(userButtonDiv);
+          } else {
+            document.getElementById("app").innerHTML = `
+              <div id="sign-in"></div>
+            `;
 
-  ```html {13}
-  <div id="user-button"></div>
-  <script>
-    const script = document.createElement('script');
-    script.setAttribute('data-clerk-publishable-key', `{{pub_key}}`);
-    script.async = true;
-    script.src = `https://[your-domain].clerk.accounts.dev/npm/@clerk/clerk-js@latest/dist/clerk.browser.js`;
+            const signInDiv =
+              document.getElementById("sign-in");
 
-    script.addEventListener('load', async function () {
-      await window.Clerk.load();
+            clerk.mountSignIn(signInDiv);
+          }
+          ```
+          </Tab>
 
-      const userButtonComponent = document.querySelector('#user-button');
+          <Tab>
+          ```html filename="index.html"
+          <!doctype html>
+          <html lang="en">
+            <head>
+              <meta charset="UTF-8" />
+              <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+              <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+              <title>Vite App</title>
+            </head>
+            <body>
+              <div id="app"></div>
+              <script type="module" src="/main.js"></script>
```

Comment:
this is how vite sets the script when you scaffold the vite project. so unless we want to write a section saying "now move your script tag to the head and add the defer prop, because defer does this..." then we should probably follow to what the scaffold does for the user. remember that this is supposed to follow the user's story, so anything we change in the code, we have to explain